scripts/build.sh

```diff
@@ -70,19 +70,47 @@ ubuntu_setup () {
         SOLVER_FLAGS="$SOLVER_FLAGS \
             -DENABLE_Z3=On -DZ3_DIR=/usr \
             -DENABLE_GOTO_CONTRACTOR=OFF \
+            -DENABLE_BITWUZLA=OFF \
         "
         return
     fi
 
     sudo apt-get update &&
     sudo apt-get install -y $PKGS &&
 
+    # Install GMP 6.3.0 from source (needed for bitwuzla)
+    echo "Installing GMP 6.3.0 from source..." &&
+    ORIGINAL_DIR="$PWD" &&
+    cd /tmp &&
+    # Try multiple mirrors for reliability
```

Comment:
Isn't there an official mirror domain that will dynamically resolve to the closest mirror?